libsrc/telestrat/cputc.s

```diff
@@ -33,18 +57,35 @@ do_not_change_color_foreground:
 
     stx     OLD_BGCOLOR
 
-    dec     SCRX                 ; Dec SCRX in order to place attribute before the right position
-
     pha
     txa                          ; Swap X to A because, X contains BGCOLOR
-    ORA     #%00010000           ; Add 16 because background color is an attribute between 16 and 23. 17 is red background for example
-    BRK_TELEMON  XFWR            ; Change color on the screen (background)
+    ora     #%00010000           ; Add 16 because background color is an attribute between 16 and 23. 17 is red background for example
+
+    jsr     display_conio
     pla
 
 do_not_change_color:
-    BRK_TELEMON  XFWR            ; Macro send char to screen (channel 0)
+    ; it continues to display_conio
+
+.endproc
+
+.proc display_conio
+    ; This routine is used to displays char on screen
+    ldy     SCRX
+    sta     (ADSCR),y
+    iny
+    cpy     #SCREEN_XSIZE
+    bne     @no_inc
+    ldy     #$00
+    sty     SCRX
+    
+    jmp     update_adscr
```

Comment:
That code is supposed to wrap-around to the _next_ text row, but it stays on the same row (the `inc SCRY` is missing).

Reply:
Fixed